src/track/trackmetadatataglib.cpp

```diff
@@ -1010,44 +1146,45 @@ bool writeTrackMetadataIntoXiphComment(TagLib::Ogg::XiphComment* pTag,
         return false;
     }
 
-    writeTrackMetadataIntoTag(pTag, trackMetadata);
-
-    // Taglib does not support the update of Vorbis comments.
-    // thus, we have to remove the old comment and add the new one
+    writeTrackMetadataIntoTag(pTag, trackMetadata,
+            WRITE_TAG_OMIT_TRACK_NUMBER | WRITE_TAG_OMIT_YEAR);
 
-    pTag->removeField("ALBUMARTIST");
-    pTag->addField("ALBUMARTIST",
-            toTagLibString(trackMetadata.getAlbumArtist()));
+    // According to https://wiki.xiph.org/Field_names "TRACKTOTAL" is
+    // the proposed field name, but some applications use "TOTALTRACKS".
+    writeXiphCommentField(pTag, "TRACKNUMBER",
+            toTagLibString(trackMetadata.getTrackNumber()));
+    writeXiphCommentField(pTag, "TRACKTOTAL",
+            toTagLibString(trackMetadata.getTrackTotal()));
+    // Remove the secondary/alternative field for total tracks to prevent
+    // inconsistencies if the tags are edited again by another external
+    // application that is unaware of "TRACKTOTAL".
+    pTag->removeField("TOTALTRACKS");
```

Comment:
Potential inconsistency if Mixxx writes both "TRACKTOTAL" and "TOTALTRACKS":
1. External program reads "TOTALTRACKS" as written by Mixxx and ignores 
"TRACKTOTAL"
2. External program updates only "TOTALTRACKS" and leaves "TRACKTOTAL" untouched
3. Upon reloading Mixxx will ignore the new value of "TOTALTRACKS" and reads 
only the original value from "TRACKTOTAL"

This is what my comment is about and why I decided to remove the unofficial 
field "TOTALTRACKS" entirely when writing tags. If you don't see the value 
of that field in an external program you are not able to update it ;) Of 
course, this doesn't save you from re-creating it.

On 01/04/2016 08:27 AM, Daniel Schürmann wrote:
>
> In src/track/trackmetadatataglib.cpp 
> <https://github.com/mixxxdj/mixxx/pull/835#discussion_r48708686>:
>
> >
> > -    pTag->removeField("ALBUMARTIST");
> > -    pTag->addField("ALBUMARTIST",
> > -            toTagLibString(trackMetadata.getAlbumArtist()));
> > +    // According to https://wiki.xiph.org/Field_names "TRACKTOTAL" is
> > +    // the proposed field name, but some applications use "TOTALTRACKS".
> > +    writeXiphCommentField(pTag, "TRACKNUMBER",
> > +            toTagLibString(trackMetadata.getTrackNumber()));
> > +    writeXiphCommentField(pTag, "TRACKTOTAL",
> > +            toTagLibString(trackMetadata.getTrackTotal()));
> > +    // Remove the secondary/alternative field for total tracks to prevent
> > +    // inconsistencies if the tags are edited again by another external
> > +    // application that is unaware of "TRACKTOTAL".
> > +    pTag->removeField("TOTALTRACKS");
>
> I a third party program uses only "TOTALTRACKS", it cannot access this 
> info anymore after Mixxx touches the file.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/mixxxdj/mixxx/pull/835/files#r48708686>.
>